sound/soc/sof/ipc4-fw-reg.h

```diff
@@ -0,0 +1,151 @@
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ */
+
+#ifndef __IPC4_FW_REG_H__
+#define __IPC4_FW_REG_H__
+
+/* Reports current ROM/FW status. */
+#define SOF_IPC4_FW_STATE_MASK		GENMASK(27, 0)
+#define SOF_IPC4_FW_MODULE_MASK		GENMASK(30, 28)
+#define SOF_IPC4_FW_DSP_STATE_MASK	BIT(31)
+
+/* currently set astate */
+#define SOF_IPC4_FW_CURR_ASTATE_MASK		GENMASK(3, 0)
+#define SOF_IPC4_FW_CACHED_IMR_STATUS_MASK	BIT(4)
+#define SOF_IPC4_FW_CURRENT_FSTATE_MASK		GENMASK(7, 5)
+#define SOF_IPC4_FW_WAKE_TICK_PERIOD_MASK	GENMASK(12, 8)
+#define SOF_IPC4_FW_ACTIVE_PIPE_COUNT_MASK	GENMASK(18, 13)
+
+#define SOF_IPC4_INVALID_STREAM_POSITION	ULONG_MAX
+
+struct sof_ipc4_pipeline_registers {
+	/*
+	 * Stream start offset (LPIB) reported by mixin module allocated on pipeline attached
+	 * to Host Output Gateway when first data is being mixed to mixout module. When data
+	 * is not mixed (right after creation/after reset) value "(u64)-1" is reported.
+	 * In number of bytes.
+	 */
+	u64 stream_start_offset;
+	/*
+	 * Stream end offset (LPIB) reported by mixin module allocated on pipeline attached
+	 * to Host Output Gateway during transition from RUNNING to PAUSED. When data
+	 * is not mixed (right after creation/after reset) value "(u64)-1" is reported. When
+	 * first data is mixed then value "0"is reported.
+	 * In number of bytes.
+	 */
+	u64 stream_end_offset;
+} __packed __aligned(4);
+
+#define SOF_IPC4_PV_MAX_SUPPORTED_CHANNELS 8
+
+struct sof_ipc4_peak_volume_regs {
+	u32 peak_meter[SOF_IPC4_PV_MAX_SUPPORTED_CHANNELS];
+	u32 current_volume[SOF_IPC4_PV_MAX_SUPPORTED_CHANNELS];
+	u32 target_volume[SOF_IPC4_PV_MAX_SUPPORTED_CHANNELS];
+} __packed __aligned(4);
+
+struct sof_ipc4_llp_reading {
+	/* lower part of 64-bit LLP */
+	u32 llp_l;
+	/* upper part of 64-bit LLP */
+	u32 llp_u;
+	/* lower part of 64-bit Wallclock */
+	u32 wclk_l;
+	/* upper part of 64-bit Wallclock */
+	u32 wclk_u;
+} __packed __aligned(4);
+
+struct sof_ipc4_llp_reading_extended {
+	struct sof_ipc4_llp_reading llp_reading;
+	/* total processed data (low part) */
+	u32 tpd_low;
+	/* total processed data (high part) */
+	u32 tpd_high;
+} __packed __aligned(4);
+
+struct sof_ipc4_llp_reading_slot {
+	u32 node_id;
+	struct sof_ipc4_llp_reading reading;
+} __packed __aligned(4);
+
+/* ROM info */
+#define SOF_IPC4_FW_FUSE_VALUE_MASK		GENMASK(7, 0)
+#define SOF_IPC4_FW_LOAD_METHOD_MASK		BIT(8)
+#define SOF_IPC4_FW_DOWNLINK_IPC_USE_DMA_MASK	BIT(9)
+#define SOF_IPC4_FW_LOAD_METHOD_REV_MASK	GENMASK(11, 10)
+#define SOF_IPC4_FW_REVISION_MIN_MASK		GENMASK(15, 12)
+#define SOF_IPC4_FW_REVISION_MAJ_MASK		GENMASK(19, 16)
+#define SOF_IPC4_FW_VERSION_MIN_MASK		GENMASK(23, 20)
+#define SOF_IPC4_FW_VERSION_MAJ_MASK		GENMASK(27, 24)
+
+/* Number of pipeline registers slots in FW Regs. */
+#define SOF_IPC4_MAX_SUPPORTED_ADSP_CORES 8
+
+/* Number of host pipeline registers slots in FW Regs. */
+#define SOF_IPC4_MAX_PIPELINE_REG_SLOTS 16
```

Comment:
it's my understanding based on @mmaka1 's explanation that this is ONLY for playback streams. It doesn't seem like IPC4 can support timestamping on capture. @mmaka1 @lgirdwood can anyone confirm that?

Reply:
confirmed,  I find the pipeline reg is not allocated for capture, only for playback.

Reply:
Yes, it is only for playback. Adjusted LLP is reported for each pipeline connected to a host output DMA.